### ISSUE TITLE
RFE: SOLVER_FAVOR for version favoring

### ISSUE THREAD

**Issue by ignatenkobrain** [openSUSE/libsolv#170]
```
repo system 0 empty
repo available 0 testtags <inline>
#>=Pkg: gcc 5 1 noarch
#>=Prv: gcc
#>=Pkg: gcc 6 1 noarch
#>=Prv: gcc
#>=Pkg: build-tools 1 1 noarch
#>=Req: gcc
system unset * system

job install name build-tools
job favor pkg gcc-5-1.noarch@available
result transaction,problems <inline>
#>install build-tools-1-1.noarch@available
#>install gcc-5-1.noarch@available
```

However, it installs `gcc-6-1.noarch@available`.

**Comment by mlschroe** [comment]
SOLVER_FAVOR currently only works for alternatives (i.e. after the candidates are pruned to the best versions).

**Comment by mlschroe** [comment]
Fixed with commit 2645bc64b322460ac77d0d051de63434d4a90a0a

**Comment by ignatenkobrain** [comment]
@mlschroe This is AWESOME!

Thanks you a lot!
